src/runtime/native/TypeOffset.cs

```diff
@@ -76,7 +81,7 @@ internal static void Use(ITypeOffsets offsets)
                 int value = (int)sourceProperty.GetValue(offsets, null);
                 offsetProperty.SetValue(obj: null, value: value, index: null);
             }
-
+            Size = ((GeneratedTypeOffsets)offsets).Size;
```

Comment:
Definitely should not cast there. The whole point of the interface is that we eventually want users to be able to supply their own `ITypeOffsets` if their Python build is very special.

Reply:
Use PyTypeType.tp_basicsize as size of PyHeapTypeObject directly instead of calculate it now.